src/Microsoft.OData.Edm/Csdl/Parsing/SchemaJsonParser.cs

```diff
@@ -0,0 +1,1564 @@
+            }
+        }
+
+        public static CsdlEnumMember ParseCsdlEnumMember(string name, JsonElement enumMemberObject, JsonParserContext context)
+        {
+            // Enumeration Member Object
+            // Enumeration type members are represented as JSON object members, where the object member name is the enumeration member name and the object member value is the enumeration member value.
+            // For members of flags enumeration types a combined enumeration member value is equivalent to the bitwise OR of the discrete values.
+            // Annotations for enumeration members are prefixed with the enumeration member name.
+
+            long? value = enumMemberObject.ParseAsInteger(context);
+            return new CsdlEnumMember(name, value, context.Location());
+        }
+
+        #endregion
+
+        #region Term
+
+        /// <summary>
+        /// Parse the <see cref="JsonElement"/> to <see cref="CsdlTerm"/>.
+        /// </summary>
+        /// <param name="name">The type definition name.</param>
+        /// <param name="element">The json value to parse from.</param>
+        /// <param name="context">The parser context.</param>
+        /// <returns>the parsed csdl term.</returns>
+        internal static CsdlTerm ParseCsdlTermType(string name, JsonElement element, JsonParserContext context)
+        {
+            // Term Object
+            Debug.Assert(element.ValueKind == JsonValueKind.Object);
+            Debug.Assert(context != null);
+
+            // A term is represented as a member of the schema object whose name is the unqualified name of the term and whose value is an object.
+            // The term object MUST contain the member $Kind with a string value of Term.
+            ValidateKind(element, context, "Term", required: true);
+
+            // It MAY contain the members $Type, $Collection, $AppliesTo, $Nullable, $MaxLength, $Precision, $Scale, $SRID, and $DefaultValue, as well as $Unicode for 4.01 and greater payloads.
+            // It MAY contain annotations.
+            // Parse the term type.
+            CsdlTypeReference typeReference = CsdlJsonParseHelper.ParseCsdlTypeReference(element, context);
+
+            IList<string> appliesTo = null;
+            string defaultValue = null;
+            IList<CsdlAnnotation> termAnnotations = new List<CsdlAnnotation>();
+            element.ParseAsObject(context, (propertyName, propertyValue) =>
+            {
+                switch (propertyName)
+                {
+                    case "$Kind":
+                    case "$Type":
+                    case "$Collection":
+                    case "$Nullable":
+                    case "$MaxLength":
+                    case "$Precision":
+                    case "$Scale":
+                    case "$Unicode":
+                    case "$SRID":
+                        // skip them because they are processed
+                        break;
+
+                    case "$BaseTerm":

```

Comment:
>$BaseTerm [](start = 26, length = 9)

Should we add a (non-fatal) error for this case?

Reply:
added

---
In reply to: [443693410](https://github.com/OData/odata.net/pull/1775#discussion_r443693410) [](ancestors = 443693410)